mqtt/mqtt-bridge/src/persist/storage/mod.rs

```diff
@@ -0,0 +1,60 @@
+pub mod ring_buffer;
+pub mod serialize;
+
+use super::StorageError;
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+#[allow(dead_code)]
+pub type StorageResult<T> = Result<T, StorageError>;
+
+#[allow(dead_code)]
+#[derive(Clone, Copy, Debug)]
+pub enum FlushOptions {
+    AfterEachWrite,
+    AfterXWrites(usize),
+    AfterXBytes(usize),
+    AfterXMilliseconds(usize),
+    Off,
+}
+
+#[allow(dead_code)]
+pub struct FlushState {
+    writes: AtomicUsize,
```

Comment:
I believe you're referring to `FlushOptions`. Where`FlushState` keeps track of writes/times and being updated with every write, and needs to be thread safe.

Reply:
Yes, you are correct Vadim in that FlushState and the pointers are atomic so that the operations from StreamWakeableState can be thread safe. However, because we need to assign waker and the PublicationStore and else use StreamWakeableState currently in a Mutex these could be non-atomic. My intention is to try to spend some time to think on the interface and see if can make the ring buffer be thread safe and be able to remove locks elsewhere. 

If this proves to be a larger problem then I think we can rely on the Mutexes elsewhere and remove atomics and Mutexes here.